plugins/basic/nexus-repository-maven/src/test/java/org/sonatype/nexus/repository/maven/internal/MavenPathTest.java

```diff
@@ -149,4 +146,64 @@ public void jarSnapshotHash() {
     assertThat(coordinatesAscSha1.subordinateOf().subordinateOf(), equalTo(mavenPath.subordinateOf()));
     assertThat(coordinatesAscSha1.main(), equalTo(mavenPath.main()));
   }
+
+  @Test
+  public void locatePom() {
+    final String path = "/org/apache/maven/maven-repository-metadata/3.3.0-SNAPSHOT/maven-repository-metadata-3.3.0-20150311.160242-1.jar.sha1";
+    final MavenPath mavenPath = pathParser.parsePath(path).locatePom();
+    assertThat(mavenPath, is(notNullValue()));
+
+    assertThat(mavenPath.getPath(), equalTo(
+        "/org/apache/maven/maven-repository-metadata/3.3.0-SNAPSHOT/maven-repository-metadata-3.3.0-20150311.160242-1.pom"));
+    assertThat(mavenPath.getFileName(),
+        equalTo("maven-repository-metadata-3.3.0-20150311.160242-1.pom"));
+    assertThat(mavenPath.getHashType(), nullValue());
```

Comment:
helper method might be useful here to avoid some of the verbosity of coordinates checking?